src/building/construction.c

```diff
@@ -334,6 +335,26 @@ int building_construction_cost(void)
     return data.cost;
 }
 
+int building_construction_size(int *x, int *y)
+{
+    if (!config_get(CONFIG_UI_SHOW_CONSTRUCTION_SIZE) ||
+        !building_construction_is_updatable() ||
+        (data.type != BUILDING_CLEAR_LAND && !data.cost)) {
+        return 0;
+    }
+    *x = data.end.x - data.start.x;
+    *y = data.end.y - data.start.y;
+    if (*x < 0) {
+        *x = -*x;
+    }
+    if (*y < 0) {
+        *y = -*y;
+    }
+    (*x)++;
```

Comment:
I prefer using local variables to calculate the value and then at the very end assign them to the passed pointers, to prevent this pointer "magic".